### PR TITLE
Add realm support

### PR DESCRIPTION
**Pull request by sadym-chromium**
* Add `script.getRealms` command.
* Track `executionContextId` creating and destroying.
* Use [`Runtime.ExecutionContextDescription. uniqueId`](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#type-ExecutionContextDescription) as a BiDi realm.
* Added e2e tests.
* Don't allow empty string as a `sandbox`.

Follow up: refactoring of the BrowsingContexts #269

### CODE REVIEW

File: src/bidiMapper/domains/protocol/bidiProtocolTypes.ts [hunk 1]
```diff
@@ -402,6 +408,86 @@ export namespace Script {
...
+  export type WorkletRealmInfo = BaseRealmInfo & {
+    type: 'worklet';
+  };
+
+  const RealmTypeSchema = zod.enum([
+    'window',
+    'dedicated-worker',
+    'shared-worker',
+    'service-worker',
+    'worker',
```
Comment (thiagowfx) [inline src/bidiMapper/domains/protocol/bidiProtocolTypes.ts:465]: if the order does not matter, can we sort this list?

File: src/bidiMapper/domains/context/browsingContextProcessor.ts [hunk 2]
```diff
@@ -312,36 +313,66 @@ export class BrowsingContextProcessor {
...
+
   async process_script_callFunction(
     params: Script.CallFunctionParameters
   ): Promise<Script.CallFunctionResult> {
-    if ('realm' in params.target) {
-      throw new Error('Realm target is not implemented yet.');
-    }
-    const target: Script.ContextTarget = params.target;
+    const scriptTarget = BrowsingContextProcessor.#getScriptTarget(
+      params.target
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/browsingContextProcessor.ts:366]:
```suggestion
      params.target,
```

File: src/bidiMapper/domains/context/browsingContextImpl.ts [hunk 3]
```diff
@@ -308,8 +319,56 @@ export class BrowsingContextImpl {
           // Default execution context is set with key `null`.
           this.#sandboxToExecutionContextIdMap.set(null, params.context.id);
         }
+        if (params.context.name !== undefined) {
+          this.#sandboxToExecutionContextIdMap.set(
+            params.context.name,
+            params.context.id
```
Comment (thiagowfx) [inline src/bidiMapper/domains/context/browsingContextImpl.ts:325]:
```suggestion
            params.context.id,
```

File: tests/_helpers.py [hunk 4]
```diff
@@ -107,25 +107,41 @@ def recursive_compare(expected, actual):
...
-        f"but is {type(expected)} instead."
+def not_one_of(not_expected_list):
+    def _not_one_of(actual):
+        for not_expected in not_expected_list:
+            assert actual != not_expected
+
+    return _not_one_of
+
+
+def comppare_sorted(key_name, expected):
```
Comment (thiagowfx) [inline tests/_helpers.py:124]: typo: compare_sorted

**Review by thiagowfx — APPROVED** [review]

File: src/bidiMapper/domains/context/browsingContextProcessor.ts (hunk 2 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts:366]: Prettier doesn't think so )

File: src/bidiMapper/domains/context/browsingContextImpl.ts (hunk 3 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextImpl.ts:325]: Same. Prettier should be updated instead, but it's out of scope.

File: src/bidiMapper/domains/protocol/bidiProtocolTypes.ts (hunk 1 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/protocol/bidiProtocolTypes.ts:465]: The order is aligned with [the order in spec](https://w3c.github.io/webdriver-bidi/#type-script-RealmInfo).

File: src/bidiMapper/domains/context/browsingContextProcessor.ts (hunk 2 above)
Comment (mathiasbynens) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts:366]: This is a style discussion. If we want trailing commas for function params across multiple lines we could set `trailingComma: 'all'` in the Prettier config. But I agree that we should discuss that separately from the code changes in this PR 👍

**Pull request merged**